lib/audio.py

```diff
@@ -0,0 +1,77 @@
+__all__ = ["compute_rec_duration"]
+
+
+import logging
+import subprocess as sp
+import tempfile
+
+import soundfile as sf
+
+
+def _compute_rec_duration_convert_to_wav(rec_audio: str) -> float:
+    """
+    Tries to compute the recording duration by converting the file provided as parameter to wav.
+    Needs ffmpeg to be installed in the system. Raises `FileNotFoundError` if not present.
+
+    :return: Duration of :param:`rec_audio` converted to wav.
+    """
+    with tempfile.NamedTemporaryFile(suffix=".wav") as tmp_audio:  # Reserve a temporary file.
+        # Convert to wav.
+        sp.run(
+            [
+                "ffmpeg",
+                "-loglevel",
+                "quiet",
+                "-i",
+                rec_audio,
+                "-y",
+                tmp_audio.name,
+            ]  # Overwrite the temporary file.
+        )
+        return compute_rec_duration(tmp_audio.name)  # Now in wav format.
+
+
+def compute_rec_duration(rec_audio: str) -> float:
+    """
+    Computes the duration of a given recording in seconds.
+
+    :param str rec_audio: Path of the recording.
+        The accepted formats are mp3, wav, aac or any audio format parseable by soundfile.
+    """
+    if rec_audio.endswith("mp3"):
+        try:
+            from mutagen.mp3 import MP3
+
+            audio_length = MP3(rec_audio).info.length
```

Comment:
I think we can return this directly
```suggestion
            return MP3(rec_audio).info.length
```